tests/unit/linters/test_base.py

```diff
@@ -60,25 +62,37 @@ def test_linter_issue_string_filename_no_url(self, linter_issue):
         assert f"{issue}" == "test: foo.txt: Linter message text"
 
 
-def test_lint_ignore_api():
-    """Test projects.Lint "ignore"-related API."""
+@pytest.fixture
+def lint_ignore_data():
+    """Yaml-loaded test data for specification of ignoring linter issues.
 
-    # We want to ignore *all* issues from this linter.
-    ignore_all = "ignore_all"
-    # We want to ignore *some* issues from this linter.
-    ignore_some = "ignore_some"
-    # We want to ignore *no* issues from this linter.
-    ignore_none = "ignore_none"
+    The data specifies that all issues for linter "linter1" should be ignored, and that the
+    issues for "linter2" should be ignored if they refer to files matching "file1" and/or
+    "/lib/file2*".
+    """
+    yaml_data = textwrap.dedent(
+        """
+        ignore:
+          linter1:          # ignore all issues for "linter1"
```

Comment:
I am glad you did this yaml, I missed the part where it is not a list of one key/value dict or a scalar like:
```
lint:
  ignore:
  - library:
    - lib/libfoo.so.1
  - classic  # disables all classic
 ```
Which I think might be more user friendly

Reply:
Updated to a list. To simplify the actual implementation of the api I ended up adding a 'private' attribute that reworks the list as a dict internally.